Kai Pastor (OpenOrienteering)
- Workaround for crashed in optimized x64 builds (QTBUG-39874, tickets:#433)

diff --git a/qtbase/src/network/kernel/qnetworkproxy_win.cpp b/qtbase/src/network/kernel/qnetworkproxy_win.cpp
index da2c020..3701114 100644
--- a/qtbase/src/network/kernel/qnetworkproxy_win.cpp
+++ b/qtbase/src/network/kernel/qnetworkproxy_win.cpp
@@ -381,6 +381,10 @@ QWindowsSystemProxy::~QWindowsSystemProxy()
         ptrWinHttpCloseHandle(hHttpSession);
 }
 
+#if defined (__GNUC__)
+  #pragma GCC push_options
+  #pragma GCC optimize ("O0")
+#endif
 void QWindowsSystemProxy::init()
 {
     if (initialized)
@@ -474,6 +478,9 @@ void QWindowsSystemProxy::init()
     functional = isAutoConfig || !proxyServerList.isEmpty();
 #endif
 }
+#if defined (__GNUC__)
+  #pragma GCC pop_options
+#endif
 
 QList<QNetworkProxy> QNetworkProxyFactory::systemProxyForQuery(const QNetworkProxyQuery &query)
 {
